### PR TITLE
Add tests for send message page

### PR DESCRIPTION
**Pull request by Pinkdesu**
- Add tests for send message page

### CODE REVIEW

File: website/frontend/src/tests/pages/Send/Send.test.tsx [hunk 1]
```diff
@@ -0,0 +1,497 @@
...
+    fireEvent.change(element, { target: { value } });
+  };
+
+  it('should show loader', () => {
+    useProgramMock();
+
+    render(<SendMessagePage path="/send/message/:programId" initialEntries={[`/send/message/${PROGRAM_ID}`]} />);
+
+    expect(screen.queryByRole('form')).toBeNull();
+    expect(screen.queryByText('New message')).toBeNull();
```
Comment (nikitayutanov) [inline website/frontend/src/tests/pages/Send/Send.test.tsx:64]:
```suggestion
    expect(screen.queryByRole('form')).not.toBeInTheDocument();
    expect(screen.queryByText('New message')).not.toBeInTheDocument();;
```

File: website/frontend/src/tests/pages/Send/Send.test.tsx [hunk 2]
```diff
@@ -0,0 +1,497 @@
...
+describe('send message page tests', () => {
+  const submit = (element: Element) => {
+    fireEvent.submit(element);
+  };
+
+  const changeFieldValue = (element: Element | Node, value: string) => {
+    fireEvent.change(element, { target: { value } });
+  };
+
+  it('should show loader', () => {
```
Comment (nikitayutanov) [inline website/frontend/src/tests/pages/Send/Send.test.tsx:58]: imo it should be part of form tests, to test if loader dissapears after requests are resolved

File: website/frontend/src/tests/pages/Send/Send.test.tsx [hunk 3]
```diff
@@ -0,0 +1,497 @@
...
+    expect(calculateGas).toBeCalledWith(
+      'handle',
+      TEST_API,
+      formValues,
+      expect.any(Object),
+      undefined,
+      null,
+      PROGRAM_ID,
+      undefined
+    );
```
Comment (nikitayutanov) [inline website/frontend/src/tests/pages/Send/Send.test.tsx:165]: Don't think that we should expect gas function execution, just check for new alert and field value

File: website/frontend/src/tests/pages/Send/Send.test.tsx [hunk 4]
```diff
@@ -0,0 +1,497 @@
...
+    useProgramMock();
+
+    render(<SendMessagePage path="/send/message/:programId" initialEntries={[`/send/message/${PROGRAM_ID}`]} />);
+
+    expect(screen.queryByRole('form')).toBeNull();
+    expect(screen.queryByText('New message')).toBeNull();
+    expect(screen.getByTestId('spinner')).toBeInTheDocument();
+  });
+
+  it('test send message form without meta', async () => {
```
Comment (nikitayutanov) [inline website/frontend/src/tests/pages/Send/Send.test.tsx:68]:
```suggestion
  it('sends message to program without meta', async () => {
```

File: website/frontend/src/tests/pages/Send/Send.test.tsx [hunk 5]
```diff
@@ -0,0 +1,497 @@
...
+
+    const sendMessageBtn = screen.getByText('Send message');
+    const calculateGasBtn = screen.getByText('Calculate Gas');
+
+    expect(valueField).toBeInTheDocument();
+    expect(gasLimitField).toBeInTheDocument();
+    expect(destinationField).toBeInTheDocument();
+
+    expect(payloadTypeField).toBeNull();
+    expect(payloadTypeSwitch).toBeNull();
```
Comment (nikitayutanov) [inline website/frontend/src/tests/pages/Send/Send.test.tsx:221]:
```suggestion
    expect(payloadTypeField).not.toBeInTheDocument();
    expect(payloadTypeSwitch).not.toBeInTheDocument();
```

File: website/frontend/src/tests/pages/Send/Send.test.tsx [hunk 6]
```diff
@@ -0,0 +1,497 @@
...
+      payloadType: 'Bytes',
+      destination: 'program',
+    };
+
+    // calculate gas
+
+    fireEvent.click(calculateGasBtn);
+
+    expect(calculateGas).toBeCalledTimes(1);
+    expect(calculateGas).toBeCalledWith(
```
Comment (nikitayutanov) [inline website/frontend/src/tests/pages/Send/Send.test.tsx:254]: same, no need to test implementation

File: website/frontend/src/tests/pages/Send/Send.test.tsx [hunk 7]
```diff
@@ -0,0 +1,497 @@
...
+        message,
+        expect.any(Object),
+        expect.any(Function),
+        undefined,
+        'u32'
+      );
+    });
+  });
+
+  it('test send message form with meta', async () => {
```
Comment (nikitayutanov) [inline website/frontend/src/tests/pages/Send/Send.test.tsx:193]:
```suggestion
  it('sends message to program with meta', async () => {
```

File: website/frontend/src/tests/pages/Send/Send.test.tsx [hunk 8]
```diff
@@ -0,0 +1,497 @@
...
+      payloadType: 'u32',
+      destination: MESSAGE_ID,
+    };
+
+    // calculate gas
+
+    fireEvent.click(calculateGasBtn);
+
+    expect(calculateGas).toBeCalledTimes(1);
+    expect(calculateGas).toBeCalledWith(
```
Comment (nikitayutanov) [inline website/frontend/src/tests/pages/Send/Send.test.tsx:365]: same

File: website/frontend/src/tests/pages/Send/Send.test.tsx [hunk 9]
```diff
@@ -0,0 +1,497 @@
...
+        message,
+        expect.any(Object),
+        expect.any(Function),
+        meta,
+        undefined
+      );
+    });
+  });
+
+  it('test send reply form without meta', async () => {
```
Comment (nikitayutanov) [inline website/frontend/src/tests/pages/Send/Send.test.tsx:291]:
```suggestion
  it('sends reply to message of program without meta', async () => {
```

File: website/frontend/src/tests/pages/Send/Send.test.tsx [hunk 10]
```diff
@@ -0,0 +1,497 @@
...
+        message,
+        expect.any(Object),
+        expect.any(Function),
+        undefined,
+        'u32'
+      );
+    });
+  });
+
+  it('test reply message form with meta', async () => {
```
Comment (nikitayutanov) [inline website/frontend/src/tests/pages/Send/Send.test.tsx:402]:
```suggestion
  it('sends reply to message of program with meta', async () => {
```

File: website/frontend/src/tests/pages/Send/Send.test.tsx [hunk 11]
```diff
@@ -0,0 +1,497 @@
...
+
+    const sendMessageBtn = screen.getAllByText('Send reply')[1];
+    const calculateGasBtn = screen.getByText('Calculate Gas');
+
+    expect(valueField).toBeInTheDocument();
+    expect(gasLimitField).toBeInTheDocument();
+    expect(messageIdField).toBeInTheDocument();
+
+    expect(payloadTypeField).toBeNull();
+    expect(payloadTypeSwitch).toBeNull();
```
Comment (nikitayutanov) [inline website/frontend/src/tests/pages/Send/Send.test.tsx:430]:
```suggestion
    expect(payloadTypeField).not.toBeIntheDocument();
    expect(payloadTypeSwitch).not.toBeIntheDocument();
```

File: website/frontend/src/tests/pages/Send/Send.test.tsx [hunk 12]
```diff
@@ -0,0 +1,497 @@
...
+      payloadType: 'Bytes',
+      destination: MESSAGE_ID,
+    };
+
+    // calculate gas
+
+    fireEvent.click(calculateGasBtn);
+
+    expect(calculateGas).toBeCalledTimes(1);
+    expect(calculateGas).toBeCalledWith(
```
Comment (nikitayutanov) [inline website/frontend/src/tests/pages/Send/Send.test.tsx:461]: same

**Review by osipov-mit — APPROVED** [review]

**Pull request merged**